applications/FluidDynamicsApplication/custom_processes/weighted_divergence_calculation_process.cpp

```diff
@@ -0,0 +1,201 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Riccardo Tosi
+//
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+
+// Application includes
+#include "custom_processes/weighted_divergence_calculation_process.h"
+
+
+namespace Kratos
+{
+    /* Public functions *******************************************************/
+
+    // Constructor
+    WeightedDivergenceCalculationProcess::WeightedDivergenceCalculationProcess(ModelPart& rModelPart):
+        Process(),
+        mrModelPart(rModelPart)
+    {
+        // Set time coefficient: computations will be performed ONLY AFTER 20% of total time of the simulation
+        mTimeCoefficient = 0.2;
```

Comment:
(minor comment)
I would add the option to let the user define this coefficient (adding a second constructor). I think it could be useful being able to control this parameter when using the divergence to remesh.

Reply:
Done!